exporter/newrelicexporter/newrelic.go

```diff
@@ -98,18 +165,50 @@ func (e exporter) pushTraceData(ctx context.Context, td pdata.Traces) (int, erro
 					continue
 				}
 
-				if err := e.harvester.RecordSpan(nrSpan); err != nil {
-					errs = append(errs, err)
-					continue
-				}
+				spans = append(spans, nrSpan)
 				goodSpans++
 			}
+			batch.Spans = append(batch.Spans, spans...)
 		}
 	}
+	batches := []telemetry.PayloadEntry{&batch}
+	insertKey := e.extractInsertKeyFromHeader(ctx)
```

Comment:
We probably should check to see if the config says we should extract the key from the header before attempting to do this.

Reply:
`extractInsertKeyFromHeader` already does this check.